pkg/module/manager.go

```diff
@@ -60,7 +69,16 @@ func (m *ManagerImpl) ExecuteModule(ctx context.Context, modName string, cfg hcl
 		return nil, fmt.Errorf("module not found %q", modName)
 	}
 
-	if err := mod.Configure(ctx, cfg, execReq.Params); err != nil {
+	protoVersion, modInfo, err := m.negotiateProtocolVersions(ctx, mod, execReq.Providers, 0)
+	if err != nil {
+		return nil, fmt.Errorf("protocol negotiation failed: %w", err)
+	}
+
+	if err := mod.Configure(ctx, Info{
```

Comment:
this doesn't seem thread safe TBH, every execution should create a new module instance.